appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -11,11 +11,16 @@ import { getThemedIconPath } from './IconPath';
 import { ISiteTreeRoot } from './ISiteTreeRoot';
 
 export function validateAppSettingKey(settings: StringDictionary, newKey?: string, oldKey?: string): string | undefined {
-    newKey = newKey ? newKey.trim() : '';
-    oldKey = oldKey ? oldKey.trim().toLowerCase() : oldKey;
+    newKey = newKey ? newKey : '';
+    if (RegExp('[^\\w\\.]+').test(newKey)) {
+        return 'App setting names can only contain letters, numbers (0-9), periods ("."), and underscores ("_")';
+    }
     if (newKey.length === 0) {
-        return 'Key must have at least one non-whitespace character.';
+        return 'App setting names must have at least one character.';
     }
+
+    newKey = newKey.trim();
```

Comment:
I'm not sure if whitespace needed to be trimmed from `newKey` here. It seemed safer to keep it